File: solutions/3DPrinter.js
let readline = require('readline');

let rl = readline.createInterface(
     process.stdin, process.stdout);
     
let strings;
let data = [ ];
let testData = {
  T: 0,
  one: [],
  two: [],
  three: [],
};
let cases = [];
function parse(data){
  testData.T = Number(data[0]);
  let testNum = testData.T;
  let line = 0;
  const numCases = Number(data[line++]);

  let cases = [];
  for(let i=1; i<=numCases; i++){
        const one = data[line++].split(' ').map((str)=>Number(str));
        const two = data[line++].split(' ').map((str)=>Number(str));
        const three = data[line++].split(' ').map((str)=>Number(str));
        cases.push({i, one,two,three});
  }
  return cases;
}

//this is the algorithm implemented to determine our answer
function solve(input){
 let final = [];
 let limit = 1000000;
 let c = [input.one[0], input.two[0], input.three[0]];
 let n = [input.one[1], input.two[1], input.three[1]];
 let y = [input.one[2], input.two[2], input.three[2]];
 let k = [input.one[3], input.two[3], input.three[3]];
 
 let min_c = Math.min(...c);
 let min_n = Math.min(...n);
 let min_y = Math.min(...y);
 let min_k = Math.min(...k); 
 let quantity = min_c + min_n+ min_y + min_k;
 if (quantity == limit){
   final.push(min_c, min_n, min_y, min_k);
   final = final.join(' ');
 }else if(quantity < limit){
   final = "IMPOSSIBLE";
 }else{
    if(min_n + min_y + min_k <= limit){
      final.push((limit-(min_n + min_y + min_k)), min_n, min_y, min_k)
      final = final.join(' ');
    }else if(min_y+min_k <= limit){
      final.push(0, (limit-(min_y + min_k)), min_y, min_k)
      final = final.join(' ')
    }else if(min_k <= limit){
      final.push(0, 0, (final- min_k), min_y);
      final = final.join(' ');
    }else{
      final.push(0,0,0,limit);
      final = final.join(' ');
    }
   
 }
 
  
  console.log(`Case #${input.i}: ${final}`);

 }

function testCases(data){
  const input = parse(data)
  input.forEach((el)=>solve(el))
  
};

rl.on('line', input => data.push(input)).on('close', () => testCases(data));

File: solutions/3DPrinter.ex
defmodule Problem do 
  
   def solve do 

   one = %{c: 300000, y: 200000, m: 300000, k: 500000}
   two = %{c: 300000, y: 200000, m: 500000, k: 300000}
   three = %{c: 300000, y: 500000, m: 300000, k: 200000}
   c = Enum.min([one.c, two.c, three.c]);
   y = Enum.min([one.m, two.m, three.m]);
   m = Enum.min([one.y, two.y, three.y]);
   k = Enum.min([one.k, two.k, three.k]);
   
   if c+y+m+k == 1000000 do 
    IO.puts "Case 1: #{c,y,m,k}"
   if c+y+m+k < 1000000 do
     IO.puts "IMPOSSIBLE"
   else 
   else{
    if m + y + k <= 1000000 do
      IO.puts "Case 1: #{(1000000-(m+y+k)),y,m,k}"
   if y+k<= 1000000 do
¡      IO.puts "Case 1: #{(1000000-(y+k)),y, k}"
   if k <= 1000000 do
      final.push(0, 0, (final- min_k), min_y);
      IO.puts "Case 1: #{0,0,(1000000-k),y}"
  else
      final.push(0,0,0,limit);
      IO.puts "Case 1: #{0,0,0,100000}"

    }
   end 
   
end


Problem.solve()